WIP: adds blacklisting capabilities to mkvomsmap script

Panos512 opened the pull request:
I created two new files 'mkvomsmap_2' and `mkgridmap_2.conf` so that we can test the two implementations in parallel. 
When we verify that the new implementation works we can switch old script to the new implementation.

I have made a test private GitLab repo and hosted a file with one DN per line. The script fetches this file and excludes from the final map all the DNs that appear in the file.

We need to decide where we will put the blacklist file and then create an access token from a GitLab account that has access to that repo. The token is passed as a parameter in the script. We could probably add it in an environmental variable and distribute it through puppet to all machines.

I already tested in vocms0135 running:

`/data/srv/current/config/frontend/mkvomsmap_2 --key /data/certs/hostkey.pem --cert /data/certs/hostcert.pem -c /data/srv/current/config/frontend/mkgridmap_2.conf -o ./testout --git_token private_token`

vkuznet commented on frontend/mkvomsmap_2:
```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python
+""" mkvomsmap -o OUTPUT --cert CERT --key KEY [--vo VONAME]
```
Panos, please update docstring to reflect changes about gitlab and blacklist. The documentation should clearly explain how it is done in a script and what it will do with a blacklist file. We also need to document content of blacklist.txt file here.

vkuznet commented on frontend/mkvomsmap_2:
```diff
@@ -0,0 +1,161 @@
+try:
+  blacklisted_dns = urllib2.urlopen(gitlab_url).read().split('\n')
+except Exception:
+  blacklisted_dns = []
```
please add 
```
import traceback
traceback.print_exc()
```
to have clear message about what was wrong with gitlab_url. For instance, it can be a network problem or it can be file issue itself.

vkuznet commented on frontend/mkvomsmap_2:
```diff
@@ -0,0 +1,161 @@
+fd, tmpname = mkstemp(dir = os.path.dirname(opts.out))
+outfile = os.fdopen(fd, "w")
+for user in users:
+  if user in blacklisted_dns:
```
I suggest to add comment here to stay explicitly that `user` is in fact is user DN.

Panos512 commented:
Thanks a lot, @vkuznet. I fixed the issues. Also renamed the argument of the script from `--git_token` to `--git-token` to follow the usual convention.

vkuznet commented:
+1 from my side, please test and deploy with @muhammadimranfarooqi to testbed first and try to blacklist yourself to see if FE will behave properly and will not allow you to access cmsweb.
